src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIClient.java

```diff
@@ -71,23 +87,20 @@ public RestAPIResponse executePost(RestAPIRequest request) throws IOException {
     HttpPost httpPost = new HttpPost(request.getUrl());
     request.getHeaders().entrySet().forEach(e -> httpPost.addHeader(e.getKey(), e.getValue()));
     httpPost.setEntity(request.getEntity());
-    RestAPIResponse apiResponse;
 
     try (CloseableHttpClient httpClient = HttpClientBuilder.create().build()) {
       try (CloseableHttpResponse httpResponse = httpClient.execute(httpPost)) {
-        apiResponse = RestAPIResponse.parse(httpResponse, request.getResponseHeaders());
+        validateHttpResponse(httpResponse);
+        try {
+          return RestAPIResponse.parse(httpResponse, request.getResponseHeaders());
+        } catch (Exception e) {
+          throw new IOException("Error while parsing JSON response", e);
+        }
       }
-    } catch (IOException e) {
-      // We're retrying all transport exceptions while executing the HTTP POST method and the generic transport
-      // exceptions in HttpClient are represented by the standard java.io.IOException class
-      // https://hc.apache.org/httpclient-legacy/exception-handling.html
-      throw e;
-    } catch (Exception e) {
-      LOG.error("Exception while executing post request", e);
-      apiResponse = RestAPIResponse.defaultErrorResponse(e.getMessage());
     }
-
-    return apiResponse;
+    // We're retrying all transport exceptions while executing the HTTP POST method and the generic transport
```

Comment:
Is this comment block still at the right place? There is no code following this block.